openapi/openapi.yml

```diff
@@ -0,0 +1,210 @@
+      responses:
+        '400':
+          description: Invalid room value
+  /room/{roomId}/peer:
+    post:
+      tags:
+        - room
+      summary: Add peer
+      operationId: addPeer
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/PeerConfig'
+      responses:
+        '200':
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Peer'
+        '500':
+          description: reached max number of peers
+  /room/{roomId}/component:
+    post:
+      tags:
+        - room
+      summary: Create server side component
+      operationId: createComponent
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/ComponentConfig'
+      responses:
+        '200':
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Component'
+        '404':
+          description: component not found
+components:
+  schemas:
+    RoomConfig:
+      type: object
+      properties:
+        maxPeers:
+          type: integer
+          format: int64
+          example: 10
+    Room:
+      type: object
+      properties: 
+        id:
+          type: integer
+          format: int64
+        config:
+          type: object
+          schema: 
+              $ref: '#/components/schemas/RoomConfig'
+        components:
+          type: array
+          items:
+            $ref: '#/components/schemas/Component'
+    ComponentConfig:
+      type: object
+      properties:
+        type:
+          $ref: '#/components/schemas/ComponentType'
+        options:
+          type:
+            object
+            
+    Component:
+      type: object
+      description: "
+        Component is a server side entity that publishes or subscribes to
+        a track. 
+        "
+      properties:
+        id:
+          type: integer
+          format: int64
+        type:
+          $ref: '#/components/schemas/ComponentType'
+    PeerConfig:
+      type: object
+      properties: 
+        type:
+          $ref: '#/components/schemas/PeerType'
+    Peer:
+      type: object
+      description: "
+        Peer is an entity that connects to the server to publish, 
+        subscribe or publish and subscribe to tracks published by producers
+        or other peers. Peer process is spawned after peer connects to the server.
+        "
+      properties:
+        id:
+          type: integer
+          format: int64
+        type:
+          $ref: '#/components/schemas/PeerType'
+    
+    PeerType:
+      type: string
+      enum:
+        - webrtc 
+
+    ComponentType:
+      type: string
+      enum:
+        - rtsp_producer
+        - file_producer
+        - hls_consumer
+        - file_consumer
```

Comment:
IMO we should introduce ComponentKind that will specify whether a component is a prodcuer or a consumer

Reply:
We already mention it in the name. What exactly do you have in mind and where do you want to use it?